third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -1184,6 +1202,81 @@ fn parse_control_exp(context: &mut Context) -> Result<(Exp, bool), Box<Diagnosti
     Ok((exp, ends_in_block))
 }
 
+// "for (iter = init; condition; update) loop_body" transforms into
+// "let iter = init; while (condition) { loop_body; update; }" and
+// "for (iter = init; condition; update; spec { ... }) loop_body" transforms into
```

Comment:
Currently we support let `for (iter = init; { spec { ... }; condition }; update) loop_body` instead.